src/gc/objecthandle.cpp

```diff
@@ -792,9 +797,18 @@ bool Ref_InitializeHandleTableBucket(HandleTableBucket* bucket, void* context)
         // No free slot.
         // Let's create a new node
         NewHolder<HandleTableMap> newMap;
-        newMap = new HandleTableMap;
+        newMap = new (nothrow) HandleTableMap;
+        if (!newMap)
+        {
+            return false;
+        }
+
+        newMap->pBuckets = new (nothrow) HandleTableBucket * [ INITIAL_HANDLE_TABLE_ARRAY_SIZE ];
+        if (!newMap->pBuckets)
+        {
+            return false;
```

Comment:
ahh, you are right of course - I did not notice the holder code above. As a side note - while looking at the usage of this, I noticed that CreateHandleStore is the one that calls this and returns nullptr if this returns false; and CreateHandleStore is called in AppDomain::Init which seems to expect things to throw OOM instead of handling nullptr returns but since CreateHandleStore doesn't throw and just returns nullptr, we should throw OOM there instead.